python/pyiceberg/io/pyarrow.py

```diff
@@ -290,7 +290,7 @@ def to_input_file(self) -> PyArrowFile:
 
 class PyArrowFileIO(FileIO):
     def __init__(self, properties: Properties = EMPTY_DICT):
-        self.get_fs: Callable[[str], FileSystem] = lru_cache(self._get_fs)
+        self.get_fs: Callable[[str], FileSystem] = lru_cache(self._initialize_fs)
```

Comment:
I'm almost inclined to remove the `lru_cache` in general and make a `fs` property. We could use a single FileIO to connect to both S3 and GCS, but I'm not sure if that happens in practice, or if we should allow it. A FileIO is bound to a table, so this would mean that a table is distributed across different object stores 🤔